### ISSUE TITLE
To integrate improvement in libusb project

### ISSUE THREAD

**Issue by mcuee** [esden/fxload#2]
If possible please help to integrate the changes from libusb fxload example.
https://github.com/libusb/libusb/blob/master/examples/fxload.c

Ref: 
https://github.com/libusb/libusb/pull/821

**Comment by esden** [comment]
The diff between the two implementations is pretty large. I don't have the time to do the merge work myself at the moment.

If you are willing to do the work of extracting the relevant improvements and creating relevant pull requests, I will be happy to review, test and merge them here.

Just FYI this is essentially just a fork of the [hotplug fxload from sourceforge](http://downloads.sourceforge.net/project/linux-hotplug/fxload/2008_10_13/fxload-2008_10_13.tar.gz) plus a bunch of improvements that I needed. I do not know if the original is being maintained or not.

**Comment by mcuee** [comment]
@esden Fair enough. It seems to me the original version is no longer being maintained.

@mriesch-wv
If you are keen, please help to create the pull request. Thanks.

**Comment by mriesch-wv** [comment]
OK. I cannot tell whether I will have time during the next months, but maybe I'll have a shot at reconciling the different versions.
